src/ZeekPluginBootstrap.cmake.in

```diff
@@ -0,0 +1,15 @@
+# This script is installed alongside ZeekPlugin.cmake for passing variables from
+# the CMake configuration step to plugins.
+
+# Allows scripts such as ZeekPlugin.cmake to locate the Zeek CMake package.
+set(ZEEK_CMAKE_PACKAGE_DIR "@ZEEK_CMAKE_INSTALL_DIR@"
+    CACHE PATH "Path the CMake package for Zeek." FORCE)
+
+# Allows scripts to locate files in the Zeek install tree.
+set(ZEEK_CMAKE_INSTALL_PREFIX "@CMAKE_INSTALL_PREFIX@"
```

Comment:
I want to suggest `ZEEK_INSTALL_PREFIX` might be enough as a name here (possibly nicer to use in plugins downstream and more in line with ZEEK_DIST, ZEEK_PLUGIN_SCRIPTS_PATH or ZEEK_SOURCE_DIR).  Have confused it about 3 times with the `ZEEK_CMAKE_INSTALL_DIR` variable above now.

There's now a `ZeekPluginConfig.cmake` and `ZeekPluginBootstrap.cmake`. My understanding is that the former is for building plugins only from the source tree and isn't actually installed and the latter for the install tree case (with quite some additional logic in ZeekPlugin.cmake to make the latter work). I wonder a bit if that could all have been centralized in ZeekPluginConfig.cmake, but have also accepted that our cmake setup is mostly over my head ;-)

Edit: I've tested with ZeekJS and there's now an issue that the CMAKE_MODULE_PATH isn't extended like before, details in #development on Slack.

Reply:
> possibly nicer to use in plugins downstream

Just to clarify, this variable is *not* intended for downstream use. I'd like to keep the name, because it really is the `CMAKE_INSTALL_DIR` that has been used to configure Zeek...

> confused it about 3 times with the `ZEEK_CMAKE_INSTALL_DIR` variable above now.

... but I agree that these two are too similar. Renaming the latter to `ZEEK_CMAKE_CONFIG_DIR`. IMHO is more descriptive. Better?

Pulling in these comments from chat:

> - From all I understand, the previous Zeek versions added <plugin-src>/cmake automatically to `CMAKE_MODULE_PATH`
> - build/compile_commands.json isn't generated within an external plugin's build directory anymore
> - When building Zeek with DEBUG enabled, previously a plugin's compile flags would also be extended with `-DDEBUG`.

They should be fixed. When I build Zeek in debug mode, I can now see `-DDEBUG` in the compile database (which is forced now as well). I do extend `CMAKE_MODULE_PATH` now, but I'm not sure if we force it into the right scope. Mind checking on your end?

Reply:
> ... but I agree that these two are too similar. Renaming the latter to ZEEK_CMAKE_CONFIG_DIR. IMHO is more descriptive. Better?

Sure, renaming the other one works too. I like the `ZEEK_CMAKE_CONFIG_DIR` (and also that it aligns with the name in the bootstrap file).

> They should be fixed. When I build Zeek in debug mode, I can now see -DDEBUG in the compile database (which is forced now as well). I do extend CMAKE_MODULE_PATH now, but I'm not sure if we force it into the right scope. Mind checking on your end?

All appear fixed. Thanks! :-)